tests/integration/profile_test.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+import os
+
+
+from contextlib import contextmanager
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+
+import pytest
+
+from .testlib import cmd as libcmd
+
+
+DUMMY0_IFNAME = 'dummy0'
+
+NMCLI_CON_ADD_CMD = [
+    'nmcli',
+    'con',
+    'add',
+    'type',
+    'dummy',
+    'con-name',
+    'testProfile',
+    'connection.autoconnect',
+    'no',
+    'ifname',
+    DUMMY0_IFNAME,
+]
+
+PROFILE_DIRECTORY = '/etc/NetworkManager/system-connections/'
+
+
+def test_delete_inactive_profiles(dummy_prepare):
+    with dummy_interface(dummy_prepare):
+        profile_exists = _profile_exists('testProfile.nmconnection')
+        assert not profile_exists
+
+
+def _profile_exists(profile_name):
+    return os.path.isfile(PROFILE_DIRECTORY + profile_name)
+
+
+@contextmanager
+def dummy_interface(ifname):
+    try:
+        desired_state = {
+            Interface.KEY: [
+                {
+                    Interface.NAME: ifname,
+                    Interface.TYPE: InterfaceType.DUMMY,
+                    Interface.STATE: InterfaceState.UP,
+                }
+            ]
+        }
+        libnmstate.apply(desired_state)
+        yield
+    finally:
+        desired_state[Interface.KEY][0][Interface.STATE] = InterfaceState.DOWN
+        libnmstate.apply(desired_state)
+
+
+@pytest.fixture
+def dummy_prepare():
+    libcmd.exec_cmd(NMCLI_CON_ADD_CMD)
+    profile_exists = _profile_exists('testProfile.nmconnection')
+    assert profile_exists
+
+    return DUMMY0_IFNAME
```

Comment:
You assume here that the test will pass. If it does not, we are left with the profile as a leftover.
There is a need to cleanup here:
try: yield ... finally: libcmd.exec_cmd(...)

Reply:
Sure, fixed.